remediation-service/handler/get_action_event_handler_test.go

```diff
@@ -24,6 +25,35 @@ func newGetActionTriggeredEvent(filename string) models.KeptnContextExtendedCE {
 	return event
 }
 
+// This test reproduces integration test self-healing until line 176 :
+// it triggers a remediation that fails because no remediation.yaml is available yet
+// remediation-service finished with result fail
+// message is "Could not get remediation.yaml file ..."
+// require.Equal(t, keptnv2.StatusErrored, finishedEventData.Status)
+// require.Equal(t, keptnv2.ResultFailed, finishedEventData.Result)
+
+func Test_Fail_Missing_YAML(t *testing.T) {
+	fakeKeptn := sdk.NewFakeKeptn("test-remediation-svc")
+	fakeKeptn.SetResourceHandler(sdk.FailingResourceHandler{})
+	fakeKeptn.AddTaskHandler("sh.keptn.event.get-action.triggered", handler.NewGetActionEventHandler())
+	fakeKeptn.NewEvent(newGetActionTriggeredEvent("test/events/get-action.triggered-0.json"))
+
+	fakeKeptn.AssertNumberOfEventSent(t, 2)
+	fakeKeptn.AssertSentEventType(t, 0, keptnv2.GetStartedEventType("get-action"))
+	fakeKeptn.AssertSentEventType(t, 1, keptnv2.GetFinishedEventType("get-action"))
+
+	fakeKeptn.AssertSentEventStatus(t, 1, keptnv2.StatusErrored)
+	fakeKeptn.AssertSentEventResult(t, 1, keptnv2.ResultFailed)
+	fakeKeptn.AssertSentEvent(t, 1, func(ce models.KeptnContextExtendedCE) bool {
+		getActionFinishedData := keptnv2.GetActionFinishedEventData{}
+		ce.DataAs(&getActionFinishedData)
+		return strings.Contains(getActionFinishedData.Message, "Could not get remediation.yaml")
+	})
+
+}
+
+// this test already checks for a passing path and for retrieving an action to set EnablePromotion to value off
+// component test should not care about action.triggered event since this depends on unleash service
```

Comment:
i think we can remove the mentioning of the unleash-service - otherwise one might think that the remediation-service only works in combination with the unleash-service, while it should not care at all which service will then handle the action.triggered event